### PR TITLE
[Not-Merge][AMD] GLM-5 performance optimization

### PR DESCRIPTION
**Pull request by wufann**
## Motivation
Optimize GLM-5 model performance.

## Modifications
1. Pad IO tensor shape in framework level for Aiter sparse attn when nhead=8 per device (eg. glm-5 TP8).
2. Delete 3 hardcode parameters in Aiter deepgemm_fp8_paged_mqa_logits api, and kernel will configure optimized parameters internally.
3. Tune GLM-5 Moe gate/router gemm_a16w16_atomic triton kernel with shape (N=256, K=6144) in Aiter.https://github.com/ROCm/aiter/pull/2213/

## Accuracy Tests
```
MODEL=/data/models/GLM-5-FP8
#MODEL=/data/models/DeepSeek-V3.2-Exp
python3 -m sglang.launch_server --model $MODEL \
    --tp 8 \
    --port 10086 \
    --trust-remote-code \
    --nsa-prefill-backend aiter\
    --nsa-decode-backend aiter \
    --disable-radix-cache

python3 /sgl-workspace/sglang/benchmark/gsm8k/bench_sglang.py --num-shots 20 --num-questions 1309 --port 10086 --parallel 96

GLM-5-FP8:
Accuracy: 0.952
Invalid: 0.000

DeekSeek-V3.2-EXP
Accuracy: 0.954
Invalid: 0.000
```

## Benchmarking and Profiling
```
Benchmark on MI350

CON="8 16 32 64"
COMBINATIONS=("8192/1024")
for combo in "${COMBINATIONS[@]}"; do
   IFS="/" read -r isl osl <<< "$combo"
   for con in $CON; do
       prompts=$(($con))
       echo "[RUNNING] prompts $prompts isl $isl osl $osl con $con"
       python3 /sgl-workspace/sglang/python/sglang/bench_serving.py \
       --backend sglang \
       --host 127.0.0.1 \
       --port 10086 \
       --dataset-name random \
       --random-range-ratio 1 \
       --num-prompt $((con * 10)) \
       --random-input $isl \
       --random-output $osl \
       --max-concurrency $con
    done
done
```

GLM-5-FP8_performance_comparison

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
4. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
5. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
6. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by HaiShaw — APPROVED** [review]

**Review by HaiShaw — COMMENTED** [review]
Merge later, see https://github.com/sgl-project/sglang/issues/21302

**Comment by HaiShaw** [comment]
@amd-bot ci-status
